js/src/admin/components/UserListPage.tsx

```diff
@@ -0,0 +1,378 @@
+    }
+
+    const columns: (ColumnData & { itemName: string })[] = this.columns().toArray();
+
+    return [
+      <p class="UserListPage-totalUsers">{app.translator.trans('core.admin.users.total_users', { count: this.userCount })}</p>,
+      <section
+        class={classList(['UserListPage-grid', this.isLoadingPage ? 'UserListPage-grid--loadingPage' : 'UserListPage-grid--loaded'])}
+        style={`grid-template-columns: repeat(${columns.length}, minmax(max-content, 300px))`}
+      >
+        {/* Render columns */}
+        {columns.map((column) => (
+          <div class="UserListPage-grid--header">{column.name}</div>
+        ))}
+
+        {/* Render user data */}
+        {this.pageData.map((user, rowIndex) =>
+          columns.map((col) => {
+            const columnContent = col.content && col.content(user);
+
+            return (
+              <div
+                class={classList(['UserListPage-grid--rowItem', rowIndex % 2 > 0 && 'UserListPage-grid--shadedRow'])}
+                data-user-id={user.id()}
+                data-column-name={col.itemName}
+              >
+                {columnContent || app.translator.trans('core.admin.users.grid.invalid_column_content')}
+              </div>
+            );
+          })
+        )}
+
+        {/* Loading spinner that shows when a new page is being loaded */}
+        {this.isLoadingPage && <LoadingIndicator />}
+      </section>,
+      <nav class="UserListPage-gridPagination">
+        <Button
+          disabled={this.pageNumber === 0}
+          title={app.translator.trans('core.admin.users.pagination.back_button')}
+          onclick={this.previousPage.bind(this)}
+          icon="fas fa-chevron-left"
+          className="Button UserListPage-backBtn"
+        />
+        <span class="UserListPage-pageNumber">
+          {app.translator.trans('core.admin.users.pagination.page_counter', {
+            current: this.pageNumber + 1,
+            total: this.getTotalPageCount(),
+          })}
+        </span>
+        <Button
+          disabled={!this.moreData}
+          title={app.translator.trans('core.admin.users.pagination.next_button')}
+          onclick={this.nextPage.bind(this)}
+          icon="fas fa-chevron-right"
+          className="Button UserListPage-nextBtn"
+        />
+      </nav>,
+    ];
+  }
+
+  /**
+   * Build an item list of columns to show for each user.
+   *
+   * Each column in the list should be an object with keys `name` and `content`.
+   *
+   * `name` is a string that will be used as the column name.
+   * `content` is a function with the User model passed as the first and only argument.
+   *
+   * See `UserListPage.tsx` for examples.
+   */
+  columns(): ItemList {
+    const columns = new ItemList();
+
+    columns.add(
+      'id',
+      {
+        name: app.translator.trans('core.admin.users.grid.default_columns.id'),
+        content: (user: User) => user.id(),
+      },
+      100
+    );
+
+    columns.add(
+      'username',
```

Comment:
What about avatars 🤔 or is there a reason we wouldn't want them here

Reply:
I just thought it'd make the rows too tall to easily go through data